pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -59,6 +59,7 @@ const (
 	attrSrcPrincipal     = "source.principal"       // source identity, e,g, "cluster.local/ns/default/sa/productpage".
 	attrRequestPrincipal = "request.auth.principal" // authenticated principal of the request.
 	attrRequestAudiences = "request.auth.audiences" // intended audience(s) for this authentication information.
+	attrRequestGroups    = "request.auth.groups"    // intended group(s) for this authentication information.
```

Comment:
"request.auth.groups" is not defined in Istio attributes vocabulary. Let's use "request.auth.claims[groups]". In RBAC policy, "group" or "properties: request.auth.claims[groups]" matches to "request.auth.claims[groups]". 

We should support string array type for JWT claims in general. For example, if a use has a "permissions" claim, which is a string array. He/she can then define conditions on "request.auth.claims[permissions]" in RBAC policies.

Reply:
Hold on. We should not put in request.auth.claims neither, as we do send claims to mixer. If we put groups into that, we will need to remove it out of claims before send data to mixer.

I think it's ok to store group in a separate list. It's only available in dynamic metadata. However, the choice of name may need to be something that distinguish from regular attributes.

Reply:
As this field is only available in the dynamic metadata and the authn Envoy filter sets the group claims into the request.auth.groups field in the dynamic metadata, here the field request.auth.groups is used to read the group claims from the dynamic metada.

Reply:
Based on the discussion, groups claim is organized under "auth.derived.claims".

Reply:
https://github.com/istio/proxy/pull/1906 is created on the proxy side to organize the groups claim under "auth.derived.claims".